test/Codec/Binary/Bech32Spec.hs

```diff
@@ -111,6 +118,71 @@ spec = do
             it (T.unpack checksum) $
                 Bech32.decode checksum `shouldBe` Left expect
 
+    describe "Parsing human-readable parts from text" $ do
+
+        describe "Known-good human readable parts parse correctly." $
+            forM_ validHumanReadableParts $ \hrp ->
+                it (T.unpack hrp) $
+                    (humanReadablePartToText <$> humanReadablePartFromText hrp)
+                    `shouldBe` Right hrp
+
+        describe "Known-bad human readable parts fail to parse." $
+            forM_ invalidHumanReadableParts $ \hrp ->
+                it (T.unpack hrp) $
+                    humanReadablePartFromText hrp `shouldSatisfy` isLeft'
```

Comment:
so these two are "manual" unit tests :+1:

Reply:
Yes, that's right. :+1: 

They're just there as an extra sanity check, to make sure we really can parse "known-good" HRP strings (as included in the original specification).